.eslintrc.js

```diff
@@ -0,0 +1,112 @@
+module.exports = {
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module',
+    ecmaFeatures: {
+      jsx: true,
+    },
+    project: ['./tsconfig.json'],
+  },
+  settings: {
+    react: {
+      pragma: 'React',
+      version: 'detect',
+    },
+  },
+  extends: [
+    'semistandard',
+    'plugin:react/recommended',
+    'plugin:react-hooks/recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:jsx-a11y/recommended',
+    'plugin:prettier/recommended',
+    'airbnb-typescript',
+  ],
+  plugins: ['react', 'prettier', 'jsx-a11y', 'check-file'],
+  rules: {
+    'react-hooks/rules-of-hooks': 'error',
+    'react-hooks/exhaustive-deps': 'warn',
+    'react/prop-types': 'off',
+    'react/button-has-type': 'error',
+    'react/no-array-index-key': 'error',
+    'react/display-name': 'error',
+    'react/jsx-no-bind': 'error',
+    'import/no-extraneous-dependencies': [
+      'error',
+      { packageDir: ['.', './packages/fusion'] },
+    ],
+    '@typescript-eslint/explicit-function-return-type': 'warn',
+    /**
+     * File/Folder naming conventions
+     * */
+    'check-file/filename-naming-convention': [
+      'error',
+      {
+        '**/*.tsx': 'PASCAL_CASE',
+        '**/*.{js,ts,mts}': 'CAMEL_CASE',
+      },
+    ],
+    'check-file/folder-naming-convention': [
+      'error',
+      {
+        'packages/**/': 'CAMEL_CASE',
+        'packages/**/src/components/*': 'PASCAL_CASE',
+        'packages/**/src/pages/*': 'PASCAL_CASE',
```

Comment:
what do we mean by "pages" ?? will it have files for consts/model etc ??

Reply:
pages is a folder that can contain components. The components here are page components. 
I cannot set a rule for constants or models folders because the name of these file follow generic camelCase style